templates/environment/cf.yml

```diff
@@ -437,6 +470,109 @@ Resources:
+    Condition: DelegateDNS
+    Properties:
+      Code:
+        ZipFile: |
+          {{.ACMValidationLambda}}
+      Handler: "index.certificateRequestHandler"
+      Timeout: 600
+      MemorySize: 512
+      Role: !GetAtt 'CustomResourceRole.Arn'
+      Runtime: nodejs10.x
+
+  DNSDelegationFunction:
+    Type: AWS::Lambda::Function
+    Condition: DelegateDNS
+    Properties:
+      Code:
+        ZipFile: |
+          {{.DNSDelegationLambda}}
+      Handler: "index.domainDelegationHandler"
+      Timeout: 600
+      MemorySize: 512
+      Role: !GetAtt 'CustomResourceRole.Arn'
+      Runtime: nodejs10.x
+
+  EnvironmentHostedZone:
+    Type: "AWS::Route53::HostedZone"
+    Condition: DelegateDNS
+    Properties:
+      HostedZoneConfig:
+        Comment: !Sub "HostedZone for environment ${EnvironmentName} - ${EnvironmentName}.${ProjectName}.${ProjectDNSName}"
+      Name: !Sub ${EnvironmentName}.${ProjectName}.${ProjectDNSName}
+
+  CustomResourceRole:
+    Type: AWS::IAM::Role
+    Condition: DelegateDNS
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          -
+            Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: "DNSandACMAccess"
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "acm:ListCertificates"
+                  - "acm:RequestCertificate"
+                  - "acm:DescribeCertificate"
+                  - "acm:GetCertificate"
+                  - "acm:DeleteCertificate"
+                  - "acm:AddTagsToCertificate"
+                  - "route53:ListHostedZonesByName"
+                  - "sts:AssumeRole"
+                  - "logs:*"
+                  - "route53:ChangeResourceRecordSets"
+                  - "route53:ListResourceRecordSets"
+                  - "route53:Get*"
+                  - "route53:Describe*"
+                Resource:
+                  - "*"
+  # Adds records for this environment's hostedzone
+  # into the project's hostedzone. This lets this
+  # environment own the DNS of the it's subdomain.
+  DelegateDNSAction:
+    Condition: DelegateDNS
+    Type: Custom::DNSDelegationFunction
+    DependsOn:
+    - DNSDelegationFunction
+    - EnvironmentHostedZone
+    Properties:
+      ServiceToken: !GetAtt DNSDelegationFunction.Arn
+      DomainName: !Sub ${ProjectName}.${ProjectDNSName}
+      SubdomainName: !Sub ${EnvironmentName}.${ProjectName}.${ProjectDNSName}
+      NameServers: !GetAtt EnvironmentHostedZone.NameServers
+      RootDNSRole: !Ref ProjectDNSDelegationRole
+
+  HTTPSCert:
+    Condition: DelegateDNS
+    Type: Custom::CertificateValidationFunction
+    DependsOn:
+    - CertificateValidationFunction
+    - EnvironmentHostedZone
+    - DelegateDNSAction
```

Comment:
Does this force the `CertificateValidationFunction` to run after creating the `DNSDelegationFunction`?

Reply:
It forces this to wait until DNS delegation is doen.